### DISCUSSION TITLE
CLI command or REST API for result of DAG file parsing

### DISCUSSION THREAD

**Discussion by vidyapanchalZS** [apache/airflow#32108] [Ideas]
Hello,

We have use case where we are continuously building DAG file as per our use case. Suppose on MWAA or airflow, dag compilation / parsing can have some errors. It might be import errors or parsing errors or some other errors. To check if we have such errors, do we have any airflow cli or any rest api exposed? If not then is it possible for us to provision one?
 
Thanks in advance.

**Comment by potiuk** [comment]
Surre look for "import errors" on CLI or API. Both are available there if you look for it.

**Reply by potiuk** [reply]
When it comes to MWAA - (not standard airflow interfaces) I think checking/askin with MWAA is a good idea which of those are exposed by them.

**Reply by vidyapanchalZS** [reply]
Hello potiuk,

Thank you for your reply. but do we have rest API or CLI command for any error present at parsing DAG file by Scheduler? If not then is it possible for us to provision one.

**Reply by potiuk** [reply]
This is a question to MWAA team, not to Airflow. Airflow has both CLI and REST api - how it is exposed in MWAA, I have no knowledge - you should raise the question/request to MWAA team - this forum is for community managed Airflow. MWAA team provides Airlfow as a service and you should ask them what is availabke - as a paying customer of MWAA you should get answer to your question there.

Asking it here is wrong place to ak it.

**Reply by shubham22** [reply]
I believe we were able to address Vidya's question over email. I'm following up on making this CLI available in MWAA. We can mark this discussion answered.
